zerver/forms.py

```diff
@@ -154,16 +154,34 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
             choices=[(lang["code"], lang["name"]) for lang in get_language_list()],
         )
 
+    @override
+    def clean(self) -> None:
+        super().clean()
+        if not self.realm_creation:
+            # The validation below only needs to happen when we are
+            # creating a new realm as part of the registration process.
+            return
+
+        create_demo = self.cleaned_data.get("create_demo", False)
+        if create_demo:
+            # Subdomains for demo organizations are not set by user input.
+            return
+
+        subdomain = self.cleaned_data.get("realm_subdomain", "")
+        try:
+            check_subdomain_available(subdomain)
+        except ValidationError as e:
+            self.add_error("realm_subdomain", e)
+
```

Comment:
Is there a reason to separately override `clean` here instead of running the same logic in `clean_realm_subdomain`. Using `clean_realm_subdomain` seems preferable for this since the validation is related to it alone.

Reply:
Yes, the reason is that the validation/cleaning for `realm_subdomain` here does depend on the cleaned value of `create_demo`.

[Django docs for clean/validate form fields that depend on each other](https://docs.djangoproject.com/en/5.2/ref/forms/validation/#cleaning-and-validating-fields-that-depend-on-each-other).